iati/resources.py

```diff
@@ -241,125 +314,159 @@ def create_lib_data_path(name):
+def create_schema_path(name, version):
     """Determine the path of a Schema with the given name at the specified version of the Standard.
 
     Args:
         name (str): The name of the Schema to locate.
-        version (str): The version of the Standard to return the Schema for. Defaults to iati.version.STANDARD_VERSION_ANY.
+        version (str / int / Decimal / iati.Version): The version of the Standard to return the Schema for.
 
     Returns:
         str: The path to a file containing the specified Schema.
 
+    Raises:
+        TypeError: If the given name is of a type that cannot be a filepath.
+        TypeError: When a specified version is of a type that cannot represent an IATI version number.
+        ValueError: If the given name is a string that cannot be a useful component of a filepath.
+        ValueError: When a specified version is not a known version of the IATI Standard.
+
     Note:
         Does not check whether the specified Schema actually exists.
 
-    Warning:
-        Further exploration needs to be undertaken in how to handle multiple versions of the Standard.
-
     Todo:
-        Test this directly rather than just the indirect tests that exist at present.
+        Determine how to handle version decorators when the version argument is not first in the list. This will enable the current private function access to be removed. See #294 for more info.
 
     """
+    version = iati.version._decimalise_integer(version)  # see todo  # pylint: disable=protected-access
+    _ensure_portable_filepath(name)
+
     return path_for_version(os.path.join(PATH_SCHEMAS, '{0}'.format(name) + FILE_SCHEMA_EXTENSION), version)
 
 
+@iati.version.allow_possible_version
 @iati.version.normalise_decimals
 def folder_name_for_version(version=iati.version.STANDARD_VERSION_ANY):
     """Return the folder name for a given version of the Standard.
 
     Args:
-        version (str): The version of the Standard to return the folder path for. Defaults to iati.version.STANDARD_VERSION_ANY.
+        version (str / int / Decimal / iati.Version): The version of the Standard to return the folder path for. Defaults to iati.version.STANDARD_VERSION_ANY.
 
     Returns:
         str: The folder name for the specified version of the Standard.
 
     Raises:
-        ValueError: When a specified version is not a valid version of the IATI Standard.
-
-    Todo:
-        Extract magic string: 'version-independent'
+        TypeError: When a specified version is of a type that cannot represent an IATI version number.
+        ValueError: When a specified version is not a known version of the IATI Standard.
 
     """
+    is_major = False
+
     if version == iati.version.STANDARD_VERSION_ANY:
-        return 'version-independent'
-    elif isinstance(version, str):
+        return PATH_VERSION_INDEPENDENT
+    elif isinstance(version, (str, int)) and not isinstance(version, bool):
```

Comment:
Suggest add todo docstring to note that this logic is necessary given lack of functionality such as `Version.is_major`.

Reply:
c880a6e2bee11768294a6ffc4fc63bd5f931671b